### PR TITLE
build: adjust for Windows

### PR DESCRIPTION
**Pull request by compnerd**

### CODE REVIEW

**Comment by gmittert** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
@swift-ci please smoke test

File: cmake/modules/Utility.cmake [hunk 1]
```diff
@@ -206,19 +201,20 @@ function(add_swift_module target name deps sources additional_args)
...
-  # FIXME: Ideally, this should be passed from the swift-ci invocation.
-  if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+    # Runpath for finding Swift core libraries in the toolchain.
+    # FIXME: Ideally, this should be passed from the swift-ci invocation.
     list(APPEND DYLYB_ARGS -Xlinker -rpath -Xlinker @loader_path/../../macosx)
-  else()
+  elseif(CMAKE_SYSTEM_NAME STREQUAL Linux)
+    list(APPEND DYLYB_ARGS -Xlinker "-rpath=\\$$ORIGIN")
     list(APPEND DYLYB_ARGS -Xlinker "-rpath=\\$$ORIGIN/../../linux")
+  elseif(CMAKE_SYSTEM_NAME STREQUAL Windows)
```
Comment (jakepetroules) [inline cmake/modules/Utility.cmake]: It's fine to not do this for this patch, but in future I'd put the Windows condition above the Linux one, turn Linux into the "else" case, and then construct the last path component of the rpath argument using the lowercased form of `uname -s` - that way Linux, FreeBSD, and any other Unix-like OSes fall into the default case automatically and get something sensible.

File: cmake/modules/Utility.cmake [hunk 2]
```diff
@@ -206,19 +201,20 @@ function(add_swift_module target name deps sources additional_args)
...
+    # NOTE(compnerd) Windows does not support RPATH
```
Comment (jakepetroules) [inline cmake/modules/Utility.cmake]: I'd remove your name from this comment; it makes it look like there's a TODO here, but there isn't - Windows simply doesn't have the concept of rpaths, so there's nothing to do.

File: cmake/modules/Utility.cmake (hunk 1 above)
Comment (compnerd) [reply cmake/modules/Utility.cmake]: That definitely wouldn't work here...the rpath encodes the OS name itself.

File: cmake/modules/Utility.cmake (hunk 2 above)
Comment (compnerd) [reply cmake/modules/Utility.cmake]: Done

**Comment by compnerd** [comment]
@swift-ci please smoke test and merge

**Comment by compnerd** [comment]
@swift-ci please smoke test

**Comment by compnerd** [comment]
@swift-ci please smoke test and merge

**Comment by compnerd** [comment]
@swift-ci please smoke test

**Comment by compnerd** [comment]
@swift-ci please smoke test Linux platform

File: cmake/modules/Utility.cmake (hunk 1 above)
Comment (jakepetroules) [reply cmake/modules/Utility.cmake]: I think you might have missed part of my comment:

> [...] construct the last path component of the rpath argument using the lowercased form of uname -s - that way Linux, FreeBSD, and any other Unix-like OSes fall into the default case automatically and get something sensible.

`uname -s` will return Linux on Linux, FreeBSD on FreeBSD, etc., thus the lowercased forms of that would be "linux", "freebsd", which should be correct for all non-Apple, non-Windows platforms.

**Pull request merged**